packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -0,0 +1,626 @@
+/// at the start of the main axis and ends with the last scrollable item at the
+/// end of the main axis.
+///
+/// Instead of rendering its children on a flat plane, it renders them
+/// as if each child is broken into its own plane and that plane is
+/// perpendicularly fixed onto a cylinder which rotates along the scrolling
+/// axis.
+///
+/// This class works in 3 coordinate systems:
+///
+/// 1- The **scrollable layout coordinates**. This coordinate system is used to
+///    communicate with [ViewportOffset] and describes its children's abstract
+///    offset from the beginning of the scrollable list at (0.0, 0.0).
+///
+///    The list is scrollable from the start of the first child item to the
+///    start of the last child item.
+///
+///    Children's layout coordinates don't change as the viewport scrolls.
+///
+/// 2- The **untransformed plane's viewport painting coordinates**. Children are
+///    not painted in this coordinate system. It's an abstract intermediary used
+///    before transforming into the next cylindrical coordinate system.
+///
+///    This system is the **scrollable layout coordinates** translated by the
+///    scroll offset such that (0.0, 0.0) is the top left corner of the
+///    viewport.
+///
+///    Because the viewport is centered at the scrollable list's scroll offset
+///    instead of starting at the scroll offset, there are paintable children
+///    ~1/2 viewport length before and after the scroll offset instead of ~1
+///    viewport length after the scroll offset.
+///
+///    Children's visibility inclusion in the viewport is determined in this
+///    system regardless of the cylinder's properties such as [diameterRatio]
+///    or [perspective]. In other words, a 100px long viewport will always
+///    paint 10-11 visible 10px children if there are enough children in the
+///    viewport.
+///
+/// 3- The **transformed cylindrical space viewport painting coordinates**.
+///    Children from system 2 get their positions transformed into a cylindrical
+///    projection matrix instead of its cartesian offset with respect to the
+///    scroll offset.
+///
+///    Children in this coordinate system are painted.
+///
+///    The wheel's size and the maximum and minimum visible angles are both
+///    controlled by [diameterRatio]. Children visible in the **untransformed
+///    plane's viewport painting coordinates**'s viewport will be radially
+///    evenly laid out between the maximum and minimum angles determined by
+///    intersecting the viewport's main axis length with a cylinder whose
+///    diameter is [diameterRatio] times longer, as long as those angles are
+///    between -pi/2 and pi/2.
+///
+///    For example, if [diameterRatio] is 2.0 and this [RenderListWheelViewport]
+///    is 100.0px in the main axis, then the diameter is 200.0. And children
+///    will be evenly laid out between that cylinder's -arcsin(0.5) and
+///    arcsin(0.5) angles.
```

Comment:
why not just say `-pi/6` and `pi/6`?

Reply:
I think it's one extra layer of cognitive indirection

Reply:
Not sure what you mean... Isn't `arcsin(0.5)` more indirection than `pi/6`?

Reply:
I meant it's easier to link the causality of input: 2 -> output: arcsin(1/2). My high school trig isn't good enough to realize input: 2 -> output: pi/6 :)

Reply:
Ah. I didn't understand that the 0.5 was related to the input.

Reply:
Changed it to arcsin(1/2) to make it clearer.